Tests.Unit.Core.ApplicationServices/ApplicationServices/SystemUsage/ItSystemUsageWriteServiceTest.cs

```diff
@@ -794,7 +799,136 @@ public void Cannot_Create_With_KLEDeviations_If_RemovedKLE_Do_Not_Exist_On_Syste
+        }
+
+        [Fact]
+        public void Can_Create_With_ExternalReferences()
+        {
+            //Arrange
+            var (systemUuid, organizationUuid, transactionMock, organization, itSystem, itSystemUsage) = CreateBasicTestVariables();
+
+            SetupBasicCreateThenUpdatePrerequisites(organizationUuid, organization, systemUuid, itSystem, itSystemUsage);
+            Configure(f => f.Inject(false));
+            var externalReferences = Many<UpdatedExternalReferenceProperties>().ToList();
+            var expectedMaster = externalReferences.OrderBy(x => A<int>()).First();
+            expectedMaster.MasterReference = true;
+
+            var input = new SystemUsageUpdateParameters
+            {
+                ExternalReferences = externalReferences.FromNullable<IEnumerable<UpdatedExternalReferenceProperties>>()
+            };
+            foreach (var externalReference in externalReferences)
+                ExpectAddExternalReferenceReturns(itSystemUsage, externalReference, CreateExternalReference(externalReference));
+
+            //Act
+            var result = _sut.Create(new SystemUsageCreationParameters(systemUuid, organizationUuid, input));
+
+            //Assert
+            Assert.True(result.Ok);
+            AssertTransactionCommitted(transactionMock);
+            Assert.Equal(expectedMaster.Title, result.Value.Reference.Title);
+        }
+
+        [Fact]
+        public void Cannot_Create_With_ExternalReferences_But_No_Master()
+        {
+            //Arrange
+            var (systemUuid, organizationUuid, transactionMock, organization, itSystem, itSystemUsage) = CreateBasicTestVariables();
+
+            SetupBasicCreateThenUpdatePrerequisites(organizationUuid, organization, systemUuid, itSystem, itSystemUsage);
+            Configure(f => f.Inject(false));
+            var externalReferences = Many<UpdatedExternalReferenceProperties>().ToList();
+
+            var input = new SystemUsageUpdateParameters
+            {
+                ExternalReferences = externalReferences.FromNullable<IEnumerable<UpdatedExternalReferenceProperties>>()
+            };
+
+            //Act
+            var result = _sut.Create(new SystemUsageCreationParameters(systemUuid, organizationUuid, input));
+
+            //Assert
+            Assert.True(result.Failed);
+            Assert.Equal(OperationFailure.BadInput, result.Error.FailureType);
+            AssertTransactionNotCommitted(transactionMock);
+        }
+
+        [Fact]
+        public void Cannot_Create_With_ExternalReferences_And_Multiple_Masters()
+        {
+            //Arrange
+            var (systemUuid, organizationUuid, transactionMock, organization, itSystem, itSystemUsage) = CreateBasicTestVariables();
+
+            SetupBasicCreateThenUpdatePrerequisites(organizationUuid, organization, systemUuid, itSystem, itSystemUsage);
+            Configure(f => f.Inject(false));
```

Comment:
Kan det passe at denne bruges for at "MasterReference: bool" bliver false i alle de oprettede?

Reply:
ja det er fordi jeg bruger autofixture bagefter til at lave Many--- jeg tilføjer en comment